File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestBoolean.js
print("getBoolean")
if(typeof getBoolean !== 'undefined')
	throw "getBoolean already defined!"
getBoolean = callback("getBoolean")
getBooleanType = typeof getBoolean
print(getBooleanType)
print(getBoolean)
if(getBooleanType !== 'boolean')
	throw "getBoolean not boolean!"
callback("setBoolean",getBoolean)

print("getBooleanVector")
if(typeof getBooleanVector !== 'undefined')
	throw "getBooleanVector already defined!"
getBooleanVector = callback("getBooleanVector")
getBooleanVectorType = typeof getBooleanVector[0]
print(getBooleanVectorType)
print(getBooleanVector)
if(getBooleanVectorType !== 'boolean')
	throw "getBooleanVector not boolean!"
callback("setBooleanVector",getBooleanVector)

print("getBooleanVectorAsList")
if(typeof getBooleanVectorAsList !== 'undefined')
	throw "getBooleanVectorAsList already defined!"
getBooleanVectorAsList = callback("getBooleanVectorAsList")
getBooleanVectorAsListType = typeof getBooleanVectorAsList[0]
print(getBooleanVectorAsListType)
print(getBooleanVectorAsList)
if(getBooleanVectorAsListType !== 'boolean')
	throw "getBooleanVectorAsList not boolean!"
callback("setBooleanVectorAsList",getBooleanVectorAsList)

print("getBooleanMatrix")
if(typeof getBooleanMatrix !== 'undefined')
	throw "getBooleanMatrix already defined!"
getBooleanMatrix = callback("getBooleanMatrix")
getBooleanMatrixType = typeof getBooleanMatrix[0][0]
print(getBooleanMatrixType)
print(getBooleanMatrix)
if(getBooleanMatrixType !== 'boolean')
	throw "getBooleanMatrix not boolean!"
callback("setBooleanMatrix",getBooleanMatrix)

print("getBooleanMatrixAsList")
if(typeof getBooleanMatrixAsList !== 'undefined')
	throw "getBooleanMatrixAsList already defined!"
getBooleanMatrixAsList = callback("getBooleanMatrixAsList")
getBooleanMatrixAsListType = typeof getBooleanMatrixAsList[0][0]
print(getBooleanMatrixAsListType)
print(getBooleanMatrixAsList)
if(getBooleanMatrixAsListType !== 'boolean')
	throw "getBooleanMatrixAsList not boolean!"
callback("setBooleanMatrixAsList",getBooleanMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-scala/src/test/java/de/invesdwin/context/scala/tests/callback/ParametersAndReturnsTestBoolean.scala
System.out.println("getBoolean")
if(getBoolean != null)
	throw new Exception("getBoolean already defined!")
val getBoolean: java.lang.Boolean = callback("getBoolean")
val getBooleanType = getBoolean.getClass()
System.out.println(getBooleanType)
System.out.println(getBoolean)
if(getBooleanType != classOf[java.lang.Boolean])
	throw new Exception("getBoolean not Boolean!")
callback("setBoolean",getBoolean)

System.out.println("getBooleanVector")
if(getBooleanVector != null)
	throw new Exception("getBooleanVector already defined!")
val getBooleanVector: Array[Boolean] = callback("getBooleanVector")
val getBooleanVectorType = getBooleanVector(0).getClass()
System.out.println(getBooleanVectorType)
System.out.println(getBooleanVector)
if(getBooleanVectorType != classOf[Boolean])
	throw new Exception("getBooleanVector not Boolean!")
callback("setBooleanVector",getBooleanVector)

System.out.println("getBooleanVectorAsList")
if(getBooleanVectorAsList != null)
	throw new Exception("getBooleanVectorAsList already defined!")
val getBooleanVectorAsList: Array[Boolean] = callback("getBooleanVectorAsList")
val getBooleanVectorAsListType = getBooleanVectorAsList(0).getClass()
System.out.println(getBooleanVectorAsListType)
System.out.println(getBooleanVectorAsList)
if(getBooleanVectorAsListType != classOf[Boolean])
	throw new Exception("getBooleanVectorAsList not Boolean!")
callback("setBooleanVectorAsList",getBooleanVectorAsList)

System.out.println("getBooleanMatrix")
if(getBooleanMatrix != null)
	throw new Exception("getBooleanMatrix already defined!")
val getBooleanMatrix: Array[Array[Boolean]] = callback("getBooleanMatrix")
val getBooleanMatrixType = getBooleanMatrix(0)(0).getClass()
System.out.println(getBooleanMatrixType)
System.out.println(getBooleanMatrix)
if(getBooleanMatrixType != classOf[Boolean])
	throw new Exception("getBooleanMatrix not Boolean!")
callback("setBooleanMatrix",getBooleanMatrix)

System.out.println("getBooleanMatrixAsList")
if(getBooleanMatrixAsList != null)
	throw new Exception("getBooleanMatrixAsList already defined!")
val getBooleanMatrixAsList: Array[Array[Boolean]] = callback("getBooleanMatrixAsList")
val getBooleanMatrixAsListType = getBooleanMatrixAsList(0)(0).getClass()
System.out.println(getBooleanMatrixAsListType)
System.out.println(getBooleanMatrixAsList)
if(getBooleanMatrixAsListType != classOf[Boolean])
	throw new Exception("getBooleanMatrixAsList not Boolean!")
callback("setBooleanMatrixAsList",getBooleanMatrixAsList)
